CMakeLists.txt

```diff
@@ -168,7 +170,18 @@ if(CMAKE_COMPILER_IS_GNU)
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wshadow")
     endif()
     if (GCC_VERSION VERSION_GREATER 5.0)
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat-signedness")
+        CHECK_C_COMPILER_FLAG("-Wformat-signedness" C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS)
+        if(C_COMPILER_SUPPORTS_WFORMAT_SIGNEDNESS)
+            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wformat-signedness")
+        endif()
+    endif()
+    CHECK_C_COMPILER_FLAG("-Wno-signed-one-bit-field" C_COMPILER_SUPPORTS_WSIGNED_ONE_BIT_FIELD)
+    if(C_COMPILER_SUPPORTS_WSIGNED_ONE_BIT_FIELD)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-signed-one-bit-field")
```

Comment:
There's one build that's failing on the CI (running Ubuntu 16.04, with GCC 5.4.0):
```
tests/scripts/all.sh -k test_null_entropy
```
Failing with:
```
…
[  9%] Building C object library/CMakeFiles/mbedcrypto.dir/entropy.c.o
…
/var/lib/build/library/entropy.c:27:2: warning: #warning "**** WARNING!  MBEDTLS_TEST_NULL_ENTROPY defined! " [-Wcpp]
 #warning "**** WARNING!  MBEDTLS_TEST_NULL_ENTROPY defined! "
  ^
/var/lib/build/library/entropy.c:28:2: warning: #warning "**** THIS BUILD HAS NO DEFINED ENTROPY SOURCES " [-Wcpp]
 #warning "**** THIS BUILD HAS NO DEFINED ENTROPY SOURCES "
  ^
/var/lib/build/library/entropy.c:29:2: warning: #warning "**** THIS BUILD IS *NOT* SUITABLE FOR PRODUCTION USE " [-Wcpp]
 #warning "**** THIS BUILD IS *NOT* SUITABLE FOR PRODUCTION USE "
  ^
[ 13%] Built target psa_constant_names_generated
[ 13%] Built target mbedtls_test
cc1: error: unrecognized command line option '-Wno-signed-one-bit-field' [-Werror]
cc1: all warnings being treated as errors
make[2]: *** [library/CMakeFiles/mbedcrypto.dir/entropy.c.o] Error 1
library/CMakeFiles/mbedcrypto.dir/build.make:638: recipe for target 'library/CMakeFiles/mbedcrypto.dir/entropy.c.o' failed
make[2]: *** Waiting for unfinished jobs....
CMakeFiles/Makefile2:400: recipe for target 'library/CMakeFiles/mbedcrypto.dir/all' failed
make[1]: *** [library/CMakeFiles/mbedcrypto.dir/all] Error 2
make: *** [all] Error 2
Makefile:138: recipe for target 'all' failed
^^^^test_null_entropy: build: default config with  MBEDTLS_TEST_NULL_ENTROPY (ASan build): command make -> 2^^^^
```
The compiler warnings are expected in this build. The compiler invocation error isn't. I don't understand why this specific build is failing, and only partially. It may be due to a pre-existing bug in the cmakefiles.